lib/puppet/provider/package/yum.rb

```diff
@@ -159,6 +172,11 @@ def install
         self.debug "Downgrading package #{@resource[:name]} from version #{is[:ensure]} to #{should}"
         operation = :downgrade
       end
+
+      if is && rpm_compareEVR(rpm_parse_evr(should), rpm_parse_evr(is[:ensure])) > 0
```

Comment:
This code is slightly confusing - it's effectively an `elsif` condition paired with line 171. I would rather have these statements joined rather than two effectively connected but separate conditionals.